dio/test/cancel_token_test.dart

```diff
@@ -0,0 +1,25 @@
+import 'package:dio/dio.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group(CancelToken, () {
+    test('cancel returns the correct DioError', () async {
+      final token = CancelToken();
+      const reason = 'cancel';
+
+      expectLater(token.whenCancel, completion((error) {
```

Comment:
I've seen multiple forms with the predicate statement in our tests, such as `equals`, `(e) => e...`, and `completion`. 🤣